app/controllers/samples_controller.rb

```diff
@@ -178,23 +179,30 @@ def dimensions
 
     # Access control enforced within samples_by_domain
     samples = samples_by_domain(domain)
-
     unless param_sample_ids.empty?
       samples = samples.where(id: param_sample_ids)
     end
-    sample_ids = samples.pluck(:id)
-
     samples = filter_samples(samples, params)
 
+    sample_ids = samples.pluck(:id)
+
     locations = samples_by_metadata_field(sample_ids, "collection_location").count
     locations = locations.map do |location, count|
       { value: location, text: location, count: count }
     end
+    not_set_count = samples.count - locations.map { |l| l[:count] }.reduce(:+)
```

Comment:
could also use .sum() right? or would that trigger a query

Reply:
Indeed. No new query, at this point it is just ruby array.